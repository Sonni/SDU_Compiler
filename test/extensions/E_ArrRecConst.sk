var arr: array of int;
var rec: record of {a: int, b: bool};

allocate arr of length 10;
allocate rec;

(* The programmer is allowed to assign as many value as wanted to, note that
   a runtime error will be thrown if array is out of bounds *)
arr = {1, 4, 3, 0, 20};

(* Every value of a record must be assigned *)
rec = {42, false};

write arr[0];
write arr[4];

write rec.a;
write rec.b;
