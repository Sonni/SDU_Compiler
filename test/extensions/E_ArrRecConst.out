1
20
42
False
